### PR TITLE
Make rate limit always display the 200s RPS

### PR DESCRIPTION
**Pull request by ldemailly**
Even if there is no 429s
```release-note
Rate limit test improvement
```

### CODE REVIEW

**Comment by ldemailly** [comment]
I'll redo this after #725 or maybe we can tweak 725 to not have a fatal at the wrong place

**Comment by ldemailly** [comment]
rebased and ready for review, PTAL

File: tests/e2e/tests/mixer/mixer_test.go [hunk 1]
```diff
@@ -569,12 +570,13 @@ func TestRateLimit(t *testing.T) {
 	got, err = vectorValue(value, map[string]string{responseCodeLabel: "200", "destination_version": "v1"})
 	if err != nil {
 		t.Logf("prometheus values for request_count:\n%s", promDump(promAPI, "request_count"))
-		fatalf(t, "Could not find successes value: %v", err)
+		t.Errorf("Could not find successes value: %v", err)
+		got = 0
 	}
 
 	got = got - prior200s
```
Comment (ldemailly) [inline tests/e2e/tests/mixer/mixer_test.go]: Note: this and the above may get negative if somehow the first call works and the second one doesn't but it's still nice for the normal case (only 200s) we've had many times when things broke

File: tests/e2e/tests/mixer/mixer_test.go [hunk 2]
```diff
@@ -516,20 +516,22 @@ func TestRateLimit(t *testing.T) {
...
-	t.Logf("Fortio Summary: %d reqs (%f 200s (%f rps), %d 400s)", totalReqs, succReqs, succReqs/opts.Duration.Seconds(), badReqs)
+	t.Logf("Fortio Summary: %d reqs (%f rps, %f 200s (%f rps), %d 400s)",
+		totalReqs, res.ActualQPS, succReqs, succReqs/actualDuration, badReqs)
 
 	// consider only successful requests (as recorded at productpage service)
 	callsToRatings := succReqs
 
 	// the rate-limit is 1 rps
-	want200s := opts.Duration.Seconds()
+	want200s := 1. * opts.Duration.Seconds()
```
Comment (mandarjog) [inline tests/e2e/tests/mixer/mixer_test.go]: instead of `int (opts.Duration.Seconds())` ? why this change?

File: tests/e2e/tests/mixer/mixer_test.go [hunk 3]
```diff
@@ -550,15 +552,16 @@ func TestRateLimit(t *testing.T) {
 	got, err := vectorValue(value, map[string]string{responseCodeLabel: "429", "destination_version": "v1"})
 	if err != nil {
 		t.Logf("prometheus values for request_count:\n%s", promDump(promAPI, "request_count"))
-		fatalf(t, "Could not find rate limit value: %v", err)
+		errorf(t, "Could not find 429s: %v", err)
+		got = 0 // want to see 200 rate even if a bug makes there is no 429s
```
Comment (mandarjog) [inline tests/e2e/tests/mixer/mixer_test.go]: Can you reword the comment.
`want to see 200 rate even if no 429s were recorded`

File: tests/e2e/tests/mixer/mixer_test.go [hunk 4]
```diff
@@ -550,15 +552,16 @@ func TestRateLimit(t *testing.T) {
...
 	}
 
 	// establish some baseline to protect against flakiness due to randomness in routing
```
Comment (mandarjog) [inline tests/e2e/tests/mixer/mixer_test.go]: Is this comment still valid? Does the test still use v2-v3 routing rule?

File: tests/e2e/tests/mixer/mixer_test.go (hunk 2 above)
Comment (ldemailly) [reply tests/e2e/tests/mixer/mixer_test.go]: I wanted to make it a bit clearer it was 1 per second

File: tests/e2e/tests/mixer/mixer_test.go (hunk 3 above)
Comment (ldemailly) [reply tests/e2e/tests/mixer/mixer_test.go]: changed

File: tests/e2e/tests/mixer/mixer_test.go (hunk 4 above)
Comment (ldemailly) [reply tests/e2e/tests/mixer/mixer_test.go]: it's not I think - tbh I think the code could be simpler and check for closer to exact 1rps 200s, rest as 429s but my goal in this PR is to show actual rates when things fail

**Comment by ldemailly** [comment]
ptal
